Hex Paper/DocumentsViewController.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  DocumentsViewController.swift
+//  Hex Paper
+//
+//  Created by Michael Johnson on 4/11/17.
+//  Copyright © 2017 Michael Johnson. All rights reserved.
+//
+
+import UIKit
+
+protocol DocumentsViewControllerDelegate {
+    func documentsViewControllerAddButtonWasTapped()
+    var documentsCount: Int { get }
+    var viewController: DocumentsViewControllerProtocol? { get set }
+}
+
+protocol DocumentsViewControllerProtocol: class {
+    func segueToDocumentScene()
+    func refreshDocumentData()
+}
+
+class DocumentsViewController: UICollectionViewController, DocumentsViewControllerProtocol {
+    var delegate: DocumentsViewControllerDelegate!
+    
+    @IBOutlet weak var addButton: UIBarButtonItem!
+
+    @IBAction func addButtonAction(_ sender: UIBarButtonItem) {
+        delegate.documentsViewControllerAddButtonWasTapped()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        delegate = DocumentsController()
+        delegate.viewController = self
+    }
```

Comment:
Should I change these lines to just `delegate = DocumentsController(viewController: self)`>